File: src/main/java/nl/crosscode/x509certbuddy/wrappers/HexDumpWrapper.java
package nl.crosscode.x509certbuddy.wrappers;

import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.security.cert.CertificateEncodingException;
import java.security.cert.X509Certificate;

public class HexDumpWrapper {

    public static String getHex(X509Certificate cert) {
        ProcessBuilder pb = new ProcessBuilder();
        pb.command("hexdump","-C");
        pb.redirectErrorStream(true);
        try {
            Process p = pb.start();
            p.getOutputStream().write(cert.getEncoded());
            p.getOutputStream().flush();
            p.getOutputStream().close();
            return new String(p.getInputStream().readAllBytes(), StandardCharsets.UTF_8);
        } catch (IOException | CertificateEncodingException e) {
            return null;
        }
    }

}

File: src/main/kotlin/nl/crosscode/x509certbuddy/wrappers/HexDumpWrapper.kt
@file:JvmName("HexDumpWrapper")
package nl.crosscode.x509certbuddy.wrappers

import nl.crosscode.x509certbuddy.ui.html.components.hex
import nl.crosscode.x509certbuddy.ui.html.components.htmlLayoutBody
import java.security.cert.X509Certificate

fun getHex(cert: X509Certificate): String {
    return htmlLayoutBody {
        hex(cert.encoded)
    }.serialize()
}
